pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryFullyQualifiedNameRule.java

```diff
@@ -141,6 +159,11 @@ private void checkImports(JavaNode node, Object data) {
         }
     }
 
+    private boolean isJavaLangImplicit(String name) {
+        // only java.lang.* is implicitly imported, but not e.g. java.lang.reflection.*
+        return name != null && name.startsWith("java.lang.") && StringUtils.countMatches(name, '.') == 2;
```

Comment:
this wouldn't flag static members such a `java.lang.ProcessBuilder.Redirect` even when doing `ProcessBuilder.Redirect` is equivalent.

Reply:
I've just updated the rule - this case is now also covered. :+1: